server/plugins/renderer/responses/pencil-response.js

```diff
@@ -137,6 +138,7 @@ class PencilResponse {
         this.data.context.in_production = false;
 
         paper.addDecorator(makeDecorator(request, this.data.context));
+        paper.setContent(this.data.renderedRegions);
```

Comment:
What?!? My mind is blown.

Reply:
We have paper helper which we use in Storefront renderer as well which takes input of an object with keys as the regionName and value as the html. So I am formatting it and just feeding it here. I was doing literal search a replace before I thought of figuring out how storefront renderer does it.

Reply:
Oh yeah sorry that was not actually a question, I was just amazed how simple it was. I know you told me before about this, just seeing it is kind of amazing.